src/promptflow/promptflow/_sdk/_service/entry.py

```diff
@@ -3,75 +3,117 @@
 from promptflow._version import VERSION
 from promptflow.exceptions import UserErrorException
 
 
+def add_start_service_action(subparsers):
+    """Add action to start pfs."""
+    start_pfs_parser = subparsers.add_parser(
+        "start",
+        description="Start promptflow service.",
+        help="pfs start",
+    )
+    start_pfs_parser.add_argument("-p", "--port", type=int, help="port of the promptflow service")
+    start_pfs_parser.add_argument(
+        "--force",
+        action="store_true",
+        help="If the port is used, the existing service will be terminated and restart a new service.",
+    )
+    start_pfs_parser.set_defaults(action="start")
+
+
+def add_show_status_action(subparsers):
+    """Add action to show pfs status."""
+    show_status_parser = subparsers.add_parser(
+        "show-status",
+        description="Display the started promptflow service info.",
+        help="pfs show-status",
+    )
+    show_status_parser.set_defaults(action="show-status")
+
+
+def start_service(args):
+    port = args.port
+    app, _ = create_app()
+    if port and is_port_in_use(port):
+        app.logger.warning(f"Service port {port} is used.")
+        raise UserErrorException(f"Service port {port} is used.")
+    if not port:
+        port = get_port_from_config(create_if_not_exists=True)
+
+    if is_port_in_use(port):
+        if args.force:
+            app.logger.warning(f"Force restart the service on the port {port}.")
+            kill_exist_service(port)
+        else:
+            app.logger.warning(f"Service port {port} is used.")
+            raise UserErrorException(f"Service port {port} is used.")
+    # Set host to localhost, only allow request from localhost.
+    app.logger.info(f"Start Prompt Flow Service on http://localhost:{port}")
+    waitress.serve(app, host="127.0.0.1", port=port)
+
+
 def main():
     command_args = sys.argv[1:]
     if len(command_args) == 1 and command_args[0] == "version":
         version_dict = {"promptflow": get_promptflow_sdk_version()}
         return json.dumps(version_dict, ensure_ascii=False, indent=2, sort_keys=True, separators=(",", ": ")) + "\n"
+    if len(command_args) == 0:
+        command_args.append("-h")
 
     if "USER_AGENT" in os.environ:
         user_agent = f"{os.environ['USER_AGENT']} local_pfs/{VERSION}"
     else:
         user_agent = f"local_pfs/{VERSION}"
     os.environ["USER_AGENT"] = user_agent
     os.environ[PF_NO_INTERACTIVE_LOGIN] = "true"
-    command_args = sys.argv[1:]
 
     parser = argparse.ArgumentParser(
         prog="pfs",
         formatter_class=argparse.RawDescriptionHelpFormatter,
         description="Prompt Flow Service",
     )
 
-    parser.add_argument("-p", "--port", type=int, help="port of the promptflow service")
     parser.add_argument(
         "-v", "--version", dest="version", action="store_true", help="show current PromptflowService version and exit"
     )
+    subparsers = parser.add_subparsers()
```

Comment:
do we have telemetry for this kind of command?

Reply:
add telemetry for this command